### PR TITLE
style:  code optimize

### LINKED ISSUE #1827: useRequest 轮询重启
当前 useRequest 轮询在 pollingInterval 由 0 变成其他值时不会启动轮询，需要手动 refresh 一次才能启动轮询，符合预期么？

看了下源码里只有停止轮询的逻辑，没有重启逻辑，是否需要添加 🤣 @crazylxr 

```diff
useUpdateEffect(() => {
  if (!pollingInterval) {
    stopPolling();
+   return () => fetchInstance.refresh()
  }
}, [pollingInterval]);
```

PS: 另外两家 react-query 跟 swr 都支持轮询重启～

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [x] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | style:  code optimize |
| 🇨🇳 Chinese | 代码优化 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: packages/hooks/src/useRequest/src/utils/subscribeFocus.ts [hunk 1]
```diff
@@ -3,9 +3,11 @@ import isBrowser from '../../../utils/isBrowser';
 import isDocumentVisible from './isDocumentVisible';
 import isOnline from './isOnline';
 
-const listeners: any[] = [];
+type Listener = (data?: any) => void;
```
Comment (brickspert) [inline packages/hooks/src/useRequest/src/utils/subscribeFocus.ts]: 这里参数应该没有 `data` 吧

File: packages/hooks/src/useRequest/src/plugins/usePollingPlugin.ts [hunk 2]
```diff
@@ -23,23 +23,26 @@ const usePollingPlugin: Plugin<any, any[]> = (
     if (!pollingInterval) {
       stopPolling();
     }
+    return () => {
+      fetchInstance.refresh();
```
Comment (brickspert) [inline packages/hooks/src/useRequest/src/plugins/usePollingPlugin.ts]: 你这个改法问题很大。

从 1=>0 也会 refresh

File: packages/hooks/src/useRequest/src/utils/subscribeFocus.ts (hunk 1 above)
Comment (li-jia-nan) [reply packages/hooks/src/useRequest/src/utils/subscribeFocus.ts]: 我改一下

File: packages/hooks/src/useRequest/src/plugins/usePollingPlugin.ts (hunk 2 above)
Comment (li-jia-nan) [reply packages/hooks/src/useRequest/src/plugins/usePollingPlugin.ts]: 我回滚一下

**Review by brickspert — APPROVED** [review]

**Pull request merged**